database/migrations/2017_12_27_082324_create_general_ledger_accounts_table.php

```diff
@@ -16,11 +16,16 @@ public function up()
         Schema::create('general_ledger_accounts', function (Blueprint $table) {
             $table->increments('id');
             $table->integer('school_id')->unsigned()->indexed();
-            $table->integer('class_group_id')->unsigned()->nullable();
             $table->string('name');
+            $table->string('slug');
+            $table->string('parent_identifier');
+            $table->integer('parent_id');
```

Comment:
add a foreign key constraint for this. i believe it will come in handy wen getting child accounts or parents

Reply:
Foreign key to itself? not never done that before. There is a helper for getting the parent in the model.